src/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Assemblies/RoAssembly.cs

```diff
@@ -107,8 +108,19 @@ public sealed override Type GetType(string name, bool throwOnError, bool ignoreC
         /// If a type is not contained or forwarded from the assembly, this method returns null (does not throw.)
         /// This supports the "throwOnError: false" behavior of Assembly.GetType(string, bool).
         /// </summary>
-        internal RoDefinitionType GetTypeCore(string ns, string name, bool ignoreCase, out Exception e) => GetTypeCore(ns.ToUtf8(), name.ToUtf8(), ignoreCase, out e);
-        internal RoDefinitionType GetTypeCore(ReadOnlySpan<byte> ns, ReadOnlySpan<byte> name, bool ignoreCase, out Exception e)
+        internal RoDefinitionType GetTypeCore(string ns, string name, bool ignoreCase, out Exception e)
+        {
+            unsafe
```

Comment:
BlobReader should probably only be used over metadata rather than pinning these strings.  Can you just have an overload that takes the array of UTF8 bytes?